unison-cli/transcripts/Transcripts.hs

```diff
@@ -7,60 +7,56 @@ module Main (main) where
 
 import Data.Bifunctor (second)
 import Data.List
-import Data.Text
-  ( pack,
-    unpack,
-  )
+import qualified Data.Text as Text
 import EasyTest
-import Shellmet () -- import instance for IsString ([Text] -> IO())
 import System.Directory
 import System.Environment (getArgs)
 import System.FilePath
-  ( splitFileName,
-    takeBaseName,
+  ( replaceExtension,
+    splitFileName,
     takeExtensions,
     (</>),
   )
 import System.IO.CodePage (withCP65001)
-import System.Process (readProcessWithExitCode)
+import Unison.Codebase.Init (withTemporaryUcmCodebase)
+import qualified Unison.Codebase.SqliteCodebase as SC
+import Unison.Codebase.TranscriptParser (TranscriptError (..), withTranscriptRunner)
 import Unison.Prelude
 
 data TestConfig = TestConfig
   { matchPrefix :: Maybe String
   }
   deriving (Show)
 
-type TestBuilder = FilePath -> FilePath -> [String] -> String -> Test ()
+type TestBuilder = FilePath -> [String] -> String -> Test ()
 
 testBuilder ::
-  FilePath -> FilePath -> [String] -> String -> Test ()
-testBuilder ucm dir prelude transcript = scope transcript $ do
-  io $ fromString ucm args
+  FilePath -> [String] -> String -> Test ()
+testBuilder dir prelude transcript = scope transcript $ do
+  outputs <- io . withTemporaryUcmCodebase SC.init "transcript" $ \(codebasePath, codebase) -> do
+    withTranscriptRunner "TODO: pass version here" Nothing $ \runTranscript -> do
+      for files $ \filePath -> do
+        transcriptSrc <- readUtf8 filePath
+        out <- runTranscript filePath transcriptSrc (codebasePath, codebase)
+        pure (filePath, out)
+  for_ outputs $ \case
+    (filePath, Left err) -> do
+      let outputFile = outputFileForTranscript filePath
+      case err of
+        TranscriptParseError msg -> do
+          crash $ "Error parsing " <> filePath <> ": " <> Text.unpack msg
+        TranscriptRunFailure errOutput -> do
+          io $ writeUtf8 outputFile errOutput
+    (filePath, Right out) -> do
+      let outputFile = outputFileForTranscript filePath
+      io $ writeUtf8 outputFile out
   ok
   where
-    files = fmap (pack . (dir </>)) (prelude ++ [transcript])
-    args = ["transcript"] ++ files
+    files = fmap (dir </>) (prelude ++ [transcript])
 
-testBuilder' ::
```

Comment:
We just use the same test-builder for both now since we don't need to inspect exit codes and/or capture stderr/stdout.